fix(deb): improve deb package

linuxgemini opened the pull request:
~~This PR is somewhat related to #1223 as currently you cannot uninstall `devolutions-gateway` properly (on Ubuntu) without modifying `/var/lib/dpkg/info/devolutions-gateway.prerm` because of the service name change making the `systemctl stop` command.~~ Post-publication edit: #1223 is a bug, hence the modifications related to this are removed.

~~And although not utilized by the service file, this PR also adds the functionality to create `/etc/defaults/devolutions-gateway` file.~~ Post-publication edit: References to the defaults file are also removed.

Lastly, the PR adds an `rm -rf /etc/devolutions-gateway` command if `postrm` is called with `purge` flag.

Post-publication edit: PR also changes the systemd reload command order and also enables (and starts) `devolutions-gateway` service to start on boot.

awakecoding commented:
Just to clarify, devolutions-agent is a different service, we haven't renamed devolutions-gateway to devolutions-agent. This being said, we may have accidentally renamed the service. @allan2 can you follow up on this?

CBenoit commented on package/Linux/gateway/debian/postinst:
```diff
@@ -3,14 +3,18 @@
 # System-wide package configuration.
 DEFAULTS_FILE="/etc/default/devolutions-gateway"
 
+if [ ! -f $DEFAULTS_FILE ]; then
+	/bin/touch $DEFAULTS_FILE
+fi
```
> And although not utilized by the service file, this PR also adds the functionality to create `/etc/defaults/devolutions-gateway` file.

This is likely something that was overlooked in the past.
I don’t think the program is ever reading the file `/etc/default/devolutions-gateway`.
The system-wide configuration is found at `/etc/devolutions-gateway/gateway.json`, and is created by the command `devolutions-gateway --config-init-only`.

CBenoit replied:
You mention that too in your original post, but I’m not sure we want to start creating this file at all given that we do not provide any option for configuring the service. We’re probably better off deleting the line above declaring `DEFAULTS_FILE`.

CBenoit commented on package/Linux/gateway/debian/postinst:
```diff
@@ -3,14 +3,18 @@
 	/bin/chmod 655 /etc/devolutions-gateway
 fi
 
 if [ -d /run/systemd/system ]; then
 	/usr/bin/devolutions-gateway service register >/dev/null
-	/usr/bin/devolutions-gateway service --config-init-only >/dev/null
-	systemctl restart devolutions-gateway >/dev/null 2>&1
 	systemctl daemon-reload
+	/usr/bin/devolutions-gateway --config-init-only >/dev/null
+	systemctl restart devolutions-agent >/dev/null 2>&1
```
This is a pertinent improvement, the correct command is `/usr/bin/devolutions-gateway --config-init-only`.
(But we should not be using the name `devolutions-agent`, the fix is not to change that here.)

CBenoit commented on package/Linux/gateway/debian/postrm:
```diff
@@ -12,6 +12,10 @@ fi
+  rm $DEFAULTS_FILE || exit 1
+fi
+
+if [ -d /etc/devolutions-gateway ]; then
+  rm -rf /etc/devolutions-gateway
 fi
```
To check.
Removing a package using `apt-get remove` should not purge the configuration. Configuration files found in `/etc`, are controlled by the system administrator, and could contain a significant amount of human effort.
However, `apt-get purge` should result in the configuration files to be deleted as well.

CBenoit commented on package/Linux/gateway/debian/prerm:
```diff
@@ -8,12 +8,13 @@ if [ "$2" = "in-favour" ]; then
-  systemctl stop devolutions-gateway >/dev/null 2>&1
+  systemctl stop devolutions-agent >/dev/null 2>&1
   /usr/bin/devolutions-gateway service unregister >/dev/null
+  systemctl daemon-reload
```
Reloading at this point sounds reasonable. :+1:

linuxgemini commented:
Heya @CBenoit !

Per your and @awakecoding's notes, I've removed the service renaming as its a bug (also renamed #1223 to reflect this).

I've also updated the code per your reviews. Thank you!

CBenoit reviewed (approved):
Everything is looking good to me. To my knowledge this is a net improvement, and I don’t see any issue. Approving and merging now to include in the next build.

@allan2 When you are back, please, double check in case I missed something. Otherwise, I think it would be good to apply a similar patch to the deb package for the agent too. Also, as discussed on Slack, it would be great to have a few sanity checks/smoke tests run on our Linux packages in the CI to prevent similar regressions in the future.

@linuxgemini Thank you for contributing this!

CBenoit commented:
@linuxgemini CI is currently broken for forks. This is fixed by https://github.com/Devolutions/devolutions-gateway/pull/1228

Do you think you could rebase your PR once https://github.com/Devolutions/devolutions-gateway/pull/1228 is merged?

Thank you!

linuxgemini commented:
@CBenoit Sure, I'll rebase when merged.
